### PR TITLE
Convert some Fetch tests to multi-global .any.js

### LINKED ISSUE #21163: Many Fetch tests do not run in service workers
Mainly due to defaulting of `.any.js` and not inserting the necessary boilerplate.

### PR DESCRIPTION
**Pull request by muodov**
Fix #21163 

There's no significant changes in the test contents, mostly just format conversion and a few minor changes (like replacing `window` with `self`).
I included only those tests that, as far as I could tell, don't add false-negatives.

The following tests I skipped:

- Window tests using DOM APIs
- Tests that introduced suspicious new failures. Mostly due to Mixed content errors and referrer differences in service workers. These tests need to be revisited one by one, and most likely can be converted with minimal changes.
  <details>
  <summary>List of files</summary>

    - /fetch/api/basic/integrity.sub.any.js
    - /fetch/api/basic/mode-no-cors.sub.any.js
    - /fetch/api/basic/mode-same-origin.any.js
    - /fetch/api/basic/referrer.any.js
    - /fetch/api/basic/response-url.sub.any.js
    - /fetch/api/basic/scheme-blob.sub.any.js
    - /fetch/api/cors/cors-redirect.any.serviceworker.html
    - /fetch/api/cors/cors-redirect-preflight.any.serviceworker.html
    - /fetch/api/cors/cors-redirect-credentials.any.serviceworker.html
    - /fetch/api/cors/cors-preflight.any.serviceworker.html
    - /fetch/api/cors/cors-preflight-status.any.serviceworker.html
    - /fetch/api/cors/cors-preflight-star.any.serviceworker.html
    - /fetch/api/cors/cors-preflight-response-validation.any.serviceworker.html
    - /fetch/api/cors/cors-preflight-referrer.any.serviceworker.html
    - /fetch/api/cors/cors-preflight-not-cors-safelisted.any.serviceworker.html
    - /fetch/api/cors/cors-preflight-cache.any.serviceworker.html
    - /fetch/api/cors/cors-origin.any.serviceworker.html
    - /fetch/api/cors/cors-origin.any.serviceworker.html
    - /fetch/api/cors/cors-no-preflight.any.serviceworker.html
    - /fetch/api/cors/cors-filtering.sub.any.serviceworker.html
    - /fetch/api/cors/cors-expose-star.sub.any.serviceworker.html
    - /fetch/api/cors/cors-cookies.any.serviceworker.html
    - /fetch/api/cors/cors-cookies-redirect.any.serviceworker.html
    - /fetch/api/cors/cors-basic.any.serviceworker.html
    - /fetch/api/redirect/redirect-mode.any.js
    - /fetch/api/redirect/redirect-origin.any.js
    - /fetch/api/redirect/redirect-referrer.any.js
    - /fetch/api/request/request-cache-only-if-cached.any.js
    - /fetch/api/request/request-clone.sub.html
    - /fetch/api/request/request-init-001.sub.html
    - fetch/api/request/request-init-003.sub.html
    - fetch/api/request/request-keepalive-quota.html
    - fetch/api/response/response-body-read-task-handling.html (uses DOM API, but seems like it doesn't have to)
    - fetch/api/response/response-consume.html (URL.createObjectURL is not available in SW)
  </details>

- There were a few occurrences of `XMLHttpRequest`, which is not available inside ServiceWorkers, so I wrapped them in `if (!self.GLOBAL.isWorker())`. Would be better to have something like `self.GLOBAL.isServiceWorker()`, but it's still better than before
- Tests that already target specific type of workers

Unfortunately, some metadata about authors and links to spec sections were lost during .html -> .any.js translation. I tried to use `// META: author=..` and `// META: help=`, but this is not accepted by linter. So I filed https://github.com/web-platform-tests/wpt/issues/26285

### CODE REVIEW

File: fetch/api/basic/accept-header.any.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // META: script=../resources/utils.js
+// META: global=window,worker
```
Comment (annevk) [inline fetch/api/basic/accept-header.any.js:2]: I suggest we put global before script. (title before global seems reasonable though.)

**Review by annevk — APPROVED** [review]
Thanks so much, this looks good to me. It's a bit unfortunate that if you ignore whitespace through GitHub's UI, it doesn't realize that more of these files are simply renames as that would make reviewing actual changes a lot easier.

If there's a next time, let's separate out formatting changes to a separate commit.

**Comment by muodov** [comment]
@annevk yeah, the changes and commit messages could have been cleaner, I will rebase.
What should I do with the stability failures? 

`/fetch/api/response/response-stream-with-broken-then.any.html` shouldn't have changed, if I understand it correctly.
`/fetch/http-cache/304-update.any.worker.html` is a new test, so theoretically it could be a new bug, but it passes 10/10 on my machine.

**Comment by annevk** [comment]
I suspect the stability failures are due to the sheer number of changes. @jgraham or @stephenmcgruer can probably override them for this PR or give specific suggestions.

**Comment by stephenmcgruer** [comment]
Phew, this is a big change! Let's dive into the failures:

[wpt-firefox-nightly-stability](https://github.com/web-platform-tests/wpt/pull/26296/checks?check_run_id=1311492051):
* `/fetch/http-cache/304-update.any.worker.html` - this appears to be a new test; [304-update.html](https://wpt.fyi/results/fetch/http-cache/304-update.html?label=master&label=experimental&aligned) exists today but that runs in the window scope not the worker scope, so we are introducing new test behavior here? In any case, the history view on wpt.fyi shows no flake in today's `fetch/http-cache/304-update.html` so I would be suspicious here that something has changed...

[wpt-chrome-dev-stability](https://github.com/web-platform-tests/wpt/pull/26296/checks?check_run_id=1311492070):
* `/fetch/api/response/response-stream-with-broken-then.any.html` - based on wpt.fyi results, this was already flaky so no action needed here.

So if you're happy with the `/fetch/http-cache/304-update.any.worker.html` case @annevk , I'm happy to admin merge for you.

**Comment by muodov** [comment]
@stephenmcgruer thanks for the double check, that's exactly what I thought! Let me do a cleanup rebase real quick and push again. Let's see if that 304 test gives different results this time.

**Comment by muodov** [comment]
@annevk @stephenmcgruer done! Sorry that you had to go through those ugly diffs, I should have rebased it from the start. After the rebase the diff makes more sense if you look commit by commit.

**Comment by annevk** [comment]
Thanks for rebasing, still looks great! And yeah, having better worker coverage was the plan and I think we have achieved that with this PR.

Please admin merge.

**Comment by zcorpan** [comment]
Merged. Thanks @muodov!

**Pull request merged**